x-pack/solutions/security/plugins/security_solution/public/detections/components/attacks/table/table_section.tsx

```diff
@@ -0,0 +1,132 @@
+}
+
+/**
+ * Renders the alerts table with grouping functionality in the attacks page.
+ */
+export const TableSection = React.memo(({ dataView }: TableSectionProps) => {
+  const getGlobalFiltersQuerySelector = useMemo(
+    () => inputsSelectors.globalFiltersQuerySelector(),
+    []
+  );
+  const globalFilters = useDeepEqualSelector(getGlobalFiltersQuerySelector);
+
+  const getGlobalQuerySelector = useMemo(() => inputsSelectors.globalQuerySelector(), []);
+  const query = useDeepEqualSelector(getGlobalQuerySelector);
+
+  const { to, from } = useGlobalTime();
+
+  const [{ loading: userInfoLoading, hasIndexWrite, hasIndexMaintenance }] = useUserData();
+
+  const { loading: listsConfigLoading } = useListsConfig();
+
+  const { showBuildingBlockAlerts, showOnlyThreatIndicatorAlerts } = useDataTableFilters(
+    TableId.alertsOnAttacksPage
+  );
+
+  // AlertsTable manages global filters itself, so not including `filters`
+  const defaultFilters = useMemo(
+    () => [
+      ...buildShowBuildingBlockFilter(showBuildingBlockAlerts),
+      ...buildThreatMatchFilter(showOnlyThreatIndicatorAlerts),
+    ],
+    [showBuildingBlockAlerts, showOnlyThreatIndicatorAlerts]
+  );
+
+  const isLoading = useMemo(
+    () => userInfoLoading || listsConfigLoading,
+    [listsConfigLoading, userInfoLoading]
+  );
+
+  const renderAlertTable = useCallback(
+    (groupingFilters: Filter[]) => {
+      return (
+        <AlertsTable
+          tableType={TableId.alertsOnAttacksPage}
+          inputFilters={[...defaultFilters, ...groupingFilters]}
+          isLoading={isLoading}
+          pageScope={PageScope.alerts} // show only detection alerts
+          disableAdditionalToolbarControls={groupingFilters.length > 0}
+        />
+      );
+    },
+    [defaultFilters, isLoading]
+  );
+
+  const groupTakeActionItems = useGroupTakeActionsItems({
+    showAlertStatusActions: Boolean(hasIndexWrite) && Boolean(hasIndexMaintenance),
+  });
+
+  const accordionExtraActionGroupStats = useMemo(
+    () => ({
+      aggregations: defaultGroupStatsAggregations,
+      renderer: defaultGroupStatsRenderer,
+    }),
+    []
+  );
+
+  const dataViewSpec = useMemo(() => {
+    return dataView.toSpec(true);
+  }, [dataView]);
+
+  return (
+    <div data-test-subj={TABLE_SECTION_TEST_ID}>
+      <GroupedAlertsTable
+        accordionButtonContent={defaultGroupTitleRenderers}
+        accordionExtraActionGroupStats={accordionExtraActionGroupStats}
+        dataView={dataView}
+        dataViewSpec={dataViewSpec}
+        defaultFilters={defaultFilters}
+        defaultGroupingOptions={defaultGroupingOptions}
+        from={from}
+        globalFilters={globalFilters}
+        globalQuery={query}
+        groupTakeActionItems={groupTakeActionItems}
+        loading={isLoading}
+        renderChildComponent={renderAlertTable}
+        tableId={TableId.alertsOnAttacksPage}
+        to={to}
+        pageScope={PageScope.attacks} // allow filtering and grouping by attack fields
```

Comment:
One risk that I can see here by using `PageScope.attacks` and use `PageScope.alerts` on line 86 above, is that we will mix dataView in the same components.
This means if one dataView has some runTime fields then we will not get them everywhere...
Right now we might be ok because we don't allow users to add runTime fields to the dataView on the `PageScope.alerts`, but this is going to change very soon. 

We need to be careful here... and I'm not exactly sure what we should do, yet.